graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -0,0 +1,349 @@
+    imputer : str
+        The type of missing value imputation to apply to the column.
+        Valid values are "mean", "median" and "most_frequent".
+    """
+
+    def __init__(
+        self, cols: Sequence[str], separator: Optional[str], normalizer: str, imputer: str
+    ) -> None:
+        assert (
+            len(cols) == 1
+        ), "DistMultiNumericalTransformation only supports one column at a time."
+        super().__init__(cols, normalizer, imputer)
+        self.multi_column = cols[0]
+
+        self.separator = separator
+        # Keep the original separator to split in pure Python
+        self.original_separator = self.separator
+        # Spark's split function uses a regexp so we need to escape
+        # special chars to be used as separators
+        if self.separator in SPECIAL_CHARACTERS:
+            self.separator = f"\\{self.separator}"
+
+    @staticmethod
+    def get_transformation_name() -> str:
+        return "DistMultiNumericalTransformation"
+
+    @staticmethod
+    def apply_norm_vector(vector_col: str, shared_norm: str, vector_df: DataFrame) -> DataFrame:
+        """
+        Applies normalizer column-wise with a single vector column as input.
+        """
+        other_cols = list(set(vector_df.columns).difference([vector_col]))
+
+        if shared_norm == "none":
+            scaled_df = vector_df
+        elif shared_norm == "min-max":
+            min_max_scaler = MinMaxScaler(inputCol=vector_col, outputCol=vector_col + "_scaled")
+
+            scaler_model = min_max_scaler.fit(vector_df)
+            scaled_df = scaler_model.transform(vector_df).drop(vector_col)
+
+            scaled_df = scaled_df.withColumnRenamed(vector_col + "_scaled", vector_col)
+            scaled_df = scaled_df.select([vector_col] + other_cols)
+        elif shared_norm == "standard":
+            col_sums_df = vector_df.select(Summarizer.sum(vector_df[vector_col]).alias("sum"))
+
+            col_sums = col_sums_df.collect()[0]["sum"]  # type: DenseVector
+
+            col_sums_array = col_sums.toArray()
+            for i, col_sum in enumerate(col_sums_array):
+                if np.isinf(col_sum) or np.isnan(col_sum) or col_sum == 0:
+                    col_sums_array[i] = 0.0
+                else:
+                    col_sums_array[i] = 1.0 / col_sums_array[i]
+
+            elwise_divider = ElementwiseProduct(
+                scalingVec=DenseVector(col_sums_array),
+                inputCol=vector_col,
+                outputCol=f"{vector_col}_scaled",
+            )
+
+            scaled_df = elwise_divider.transform(vector_df).drop(vector_col)
+
+            scaled_df = scaled_df.withColumnRenamed(vector_col + "_scaled", vector_col)
+            scaled_df = scaled_df.select([vector_col] + other_cols)
+        else:
+            raise RuntimeError(f"Unknown normalizer requested for col {vector_col}: {shared_norm}")
+
+        return scaled_df
+
+    def apply(self, input_df: DataFrame) -> DataFrame:
```

Comment:
Docstring for this func.

Reply:
The docstring is provided at the parent class. Do we want to have a different `apply` docstring for each transformation?